ocean_lib/web3_internal/contract_utils.py

```diff
@@ -31,6 +31,35 @@ def get_contract_definition(contract_name: str) -> Dict[str, Any]:
         return json.load(f)
 
 
+@enforce_types
+def get_contract_filename(contract_name: str) -> str:
```

Comment:
I think this can be simplified. Let's find and assert the path first, then go through the contracts. It will be much clearer to read. And it can be split in two different functions:
- one for getting the base location of address file and contracts (I think something similar already exists but it only accounts for the artifacts imported path and not ADDRESS_FILE env)
- one for retrieving the contract based on that location from (1)